src/sentry/notifications/notifications/rules.py

```diff
@@ -171,7 +171,7 @@ def get_context(self) -> MutableMapping[str, Any]:
 
         occurrence = getattr(self.event, "occurrence", None)
         template_name = (
-            occurrence.get("evidence_data", {}).get("template_name") if occurrence else None
+            getattr(occurrence, "evidence_data", {}).get("template_name") if occurrence else None
         )
```

Comment:
we shouldn't be guessing about these -- why are we dealing with sometimes-present attributes?

Reply:
Because we can get `Event` or `GroupEvent` in this function. If I was getting a consistent type, I wouldn't have to guess.

Reply:
I believe type inference will work better if you do `occurrence.evidence_data if isinstance(occurrence, GroupEvent) else {}`